File: doctolib/src/main/java/sopra/doctolib/model/Creneau.java
package sopra.doctolib.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.Version;

@Entity
public class Creneau {
	@Id
	@GeneratedValue
	private Long id;
	@Version
	private int version;
	@Temporal(TemporalType.DATE)
	@Column(name = "date", nullable = false)
	private Date date;
	@Column(name = "disponible", nullable = false)
	private Boolean disponible;

	@ManyToOne
	@JoinColumn(name = "lieu_id")
	private Lieu lieu;
	
	@ManyToOne
	@JoinColumn(name = "praticien_id")
	private Praticien praticien;
	
	@ManyToOne
	@JoinColumn(name = "rdv_id")
	private RendezVous rendezVous;

	public Creneau() {
		super();
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	public Date getDate() {
		return date;
	}

	public void setDate(Date date) {
		this.date = date;
	}

	public Boolean getDisponible() {
		return disponible;
	}

	public void setDisponible(Boolean disponible) {
		this.disponible = disponible;
	}

	public Lieu getLieu() {
		return lieu;
	}

	public void setLieu(Lieu lieu) {
		this.lieu = lieu;
	}

	public Praticien getPraticien() {
		return praticien;
	}

	public void setPraticien(Praticien praticien) {
		this.praticien = praticien;
	}

	public RendezVous getRendezVous() {
		return rendezVous;
	}

	public void setRendezVous(RendezVous rendezVous) {
		this.rendezVous = rendezVous;
	}

}

File: doctolib-angular/src/app/Model/creneau.ts
import {Praticien} from './praticien';
import {Lieu} from './lieu';
import {RendezVous} from './rendez-vous';

export class Creneau {
  id: number;
  version: number;
  date: Date;
  disponible: boolean;
  lieu: Lieu;
  praticien: Praticien;
  rendezVous: RendezVous;

  constructor(date?:Date, id?:number, version?:number, disponible?:boolean, lieu?: Lieu, praticien?: Praticien, rendezVous?: RendezVous) {
    this.id = id;
    this.version = version;
    this.date = date;
    this.disponible = disponible;
    this.lieu = lieu;
    this.praticien = praticien;
    this.rendezVous= rendezVous;

  }
}
